core/services/relay/evm/evmtesting/run_tests.go

```diff
@@ -34,237 +45,291 @@ func RunChainComponentsInLoopEvmTests[T TestingT[T]](t T, it ChainComponentsInte
-		triggerStaticBytes(t, it, val1, val2, val3, val4, val5, val6, val7, raw)
-		triggerStaticBytes(t, it, 1337, 7331, 4747, val4, val5, val6, val7, raw)
-		triggerStaticBytes(t, it, 7331, 4747, 1337, val4, val5, val6, val7, raw)
-		triggerStaticBytes(t, it, 4747, 1337, 7331, val4, val5, val6, val7, raw)
-
-		assert.Eventually(t, func() bool {
-			sequences, err := cr.QueryKey(ctx, boundContract, query.KeyFilter{Key: staticBytesEventName, Expressions: []query.Expression{
-				query.Comparator("msgTransmitterEvent",
-					primitives.ValueComparator{
-						Value:    dataWordOnChainValueToQuery,
-						Operator: primitives.Eq,
-					}),
+		},
+		{
+			Name: QueryKeyFilteringOnDataWordsUsingValueComparatorsOnFieldsWithManualIndex,
+			Test: func(t T) {
+				ctx := tests.Context(t)
+				cr := it.GetContractReader(t)
+				require.NoError(t, cr.Bind(ctx, it.GetBindings(t)))
+				bindings := it.GetBindings(t)
+				boundContract := BindingsByName(bindings, AnyContractName)[0]
+				require.NoError(t, cr.Bind(ctx, bindings))
+				empty12Bytes := [12]byte{}
+				val1, val2, val3, val4 := uint32(1), uint32(2), uint32(3), uint64(4)
+				val5, val6, val7 := [32]byte{}, [32]byte{6}, [32]byte{7}
+				copy(val5[:], append(empty12Bytes[:], 5))
+				raw := []byte{9, 8}
+
+				var buf []byte
+				buf = binary.BigEndian.AppendUint32(buf, val1)
+				buf = binary.BigEndian.AppendUint32(buf, val2)
+				buf = binary.BigEndian.AppendUint32(buf, val3)
+				buf = binary.BigEndian.AppendUint64(buf, val4)
+				dataWordOnChainValueToQuery := buf[:]
+
+				resExpected := append(buf, common.LeftPadBytes(val5[:], 32)...)
+				resExpected = append(resExpected, common.LeftPadBytes(val6[:], 32)...)
+				resExpected = append(resExpected, common.LeftPadBytes(val7[:], 32)...)
+				resExpected = append(resExpected, raw...)
+
+				type eventResAsStruct struct {
+					Message *[]uint8
+				}
+				wrapExpectedRes := eventResAsStruct{Message: &resExpected}
+
+				// emit the one we want to search for and a couple of random ones to confirm that filtering works
+				triggerStaticBytes(t, it, val1, val2, val3, val4, val5, val6, val7, raw)
+				triggerStaticBytes(t, it, 1337, 7331, 4747, val4, val5, val6, val7, raw)
+				triggerStaticBytes(t, it, 7331, 4747, 1337, val4, val5, val6, val7, raw)
+				triggerStaticBytes(t, it, 4747, 1337, 7331, val4, val5, val6, val7, raw)
+
+				assert.Eventually(t, func() bool {
+					sequences, err := cr.QueryKey(ctx, boundContract, query.KeyFilter{Key: staticBytesEventName, Expressions: []query.Expression{
+						query.Comparator("msgTransmitterEvent",
+							primitives.ValueComparator{
+								Value:    dataWordOnChainValueToQuery,
+								Operator: primitives.Eq,
+							}),
+					},
+					}, query.LimitAndSort{}, eventResAsStruct{})
+					return err == nil && len(sequences) == 1 && reflect.DeepEqual(wrapExpectedRes, sequences[0].Data)
+				}, it.MaxWaitTimeForEvents(), time.Millisecond*10)
```

Comment:
Pre-existing, but 10ms seems unnecessarily fast:
```suggestion
				}, it.MaxWaitTimeForEvents(), 100*time.Millisecond)
```